### ISSUE TITLE
Automatically authenticate when user executes k8s commands 

### ISSUE THREAD

**Issue by moisesmejia1200** [gravitational/teleport#18703]
What would you like Teleport to do?

Admin would like to integrate tsh with kube config, the goal is that when the user executes kubectl they will automatically authenticate first and execute the `kubectl` command

At the moment the users execute kubectl and get the following message, as they are not login.
The connection to the server localhost:8080 was refused - did you specify the right host or port?
Kubeconfig
``` 
- name: <name>
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1beta1
      args:
      - kube
      - credentials
      - --kube-cluster=<URL>
      - --teleport-cluster=<URL>
      - --proxy=<URL>
      command: /usr/local/bin/tsh
      env: null
      interactiveMode: IfAvailable
      provideClusterInfo: false
```
The customer has added the `--auth=okta` option, however as this does not return the correct json format so this can't be used to query k8s resources.
```
If browser window does not open automatically, open it by clicking on the link:
 http://127.0.0.1:49484/4ae35e72-7ae6-466b-991c-6f4907988017
{.... }
```
What problem does this solve?

This would allow users to not have to authenticate and just execute kubectl command

If a workaround exists, please include it.
Users must authenticate via tsh then query k8s resources.

**Comment by AntonAM** [comment]
@moisesmejia1200 do we know what tsh version client is using? I would suggest them to try with the latest version.
It is working as expected for me, after adding `--auth` command into kubeconfig - if cert expired it automatically relogins through SSO (I tested with github SSO). Looks like this was fixed by https://github.com/gravitational/teleport/pull/10496

As for the `The connection to the server localhost:8080 was refused - did you specify the right host or port?` message - it happens when kubectl doesn't have correct information about cluster address (so it tries default), it can happen if client does explicit `tsh logout` - we're deleting cluster information from the kubeconfig in that case. But in this case, since logout was explicit, I would say no automatic relogin when using `kubectl` is an expected behaviour.
